### DISCUSSION TITLE
how to specify config file manually

### DISCUSSION THREAD

**Discussion by bigkase** [PHPCSStandards/PHP_CodeSniffer#532] [Q&A]
Hello everyone,

When I use `--config-set` option then will be generated `CodeSniffer.conf` file.

This file automatically detected by the tool, but is there an option to manually set this file for the tool?

**Comment by jrfnl** [comment]
@bigkase No, there is not. However, everything you can set as a config option, can also be set from a custom ruleset, so if you'd prefer to have those settings in a committed file in your repo, you can.

Alternatively, if you don't want those settings committed to the repo, but do want to use them always, you could set those settings in a config-overload file, i.e. if the committed package ruleset is `phpcs.xml.dist`, you can have a `phpcs.xml` file in your project root (and set to be ignored via `.gitignore`) which contains something like:
```xml
<ruleset name="MyProject">
    <rule ref="phpcs.xml.dist"/>

    ... your custom extra settings ...
</ruleset>
```

For examples on how to set things via the ruleset, see: https://github.com/PHPCSStandards/PHP_CodeSniffer/wiki/Annotated-Ruleset
